dlt/pipeline/platform.py

```diff
@@ -0,0 +1,47 @@
+"""Implements SupportsTracking"""
+from typing import Any
+import requests
+from concurrent.futures import ThreadPoolExecutor
+
+from dlt.pipeline.trace import PipelineTrace, PipelineStepTrace, TPipelineStep, SupportsPipeline
+from dlt.common import json
+from dlt.common.runtime import logger
+
+_THREAD_POOL: ThreadPoolExecutor = None
```

Comment:
is this temporary code? we should reuse the segment pool or at least somehow add this pool to `at_exit` handler otherwise messages will not go out at the end. 

maybe extract this sending pool to common module that creates a pool and registers it in exit handler?

also implementation below does not have a method to stop the pool. won't that be a problem when testing?

Reply:
I have extracted a managedexecutor class. it does not seem a missing stop method is a problem when testing...